pkg/agent/prommetrics.go

```diff
@@ -0,0 +1,104 @@
+package agent
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/collectors"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"k8s.io/klog/v2"
+)
+
+type PromMetrics struct {
+	schedulerRequests         *prometheus.CounterVec
+	informantRequestsOutbound *prometheus.CounterVec
+	informantRequestsInbound  *prometheus.CounterVec
+}
+
+func startPrometheusServer(ctx context.Context, globalstate *agentState) (PromMetrics, error) {
```

Comment:
I kind of tend to prefer returning pointers for things like this, even though the double indirection is admittedly annoying.